src/main/java/hudson/plugins/jira/JiraRestService.java

```diff
@@ -70,12 +78,24 @@ public JiraRestService(URI uri, JiraRestClient jiraRestClient, String username,
             throw new RuntimeException("failed to encode username:password using Base64");
         }
         this.jiraRestClient = jiraRestClient;
+
+        final StringBuilder buffer = new StringBuilder();
+        if (uri.getPath() != null) {
+            buffer.append(uri.getPath());
+            if (!uri.getPath().endsWith("/")) {
+                buffer.append('/');
+            }
+        } else {
+            buffer.append('/');
+        }
+        buffer.append(BASE_API_PATH);
+        baseApiPath = buffer.toString();
```

Comment:
It will be good to have some tests explaining how baseApiPath builded with different uri's